### PR TITLE
new blog on opinions

### PR DESCRIPTION
**Pull request by mashooq**

### CODE REVIEW

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+image:
+    src: /assets/img/custom/blog/hierarchy-of-opinion.jpg
+tags:
+- craftsmanship
+
+---
+
+The word opinion has two distinct meanings:
+
+1. A view or judgement formed about something, not necessarily based on fact or knowledge.
```
Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md:17]: Please cite the source for these definitions

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+2. A statement of advice by an expert on a professional matter.
+
+Synonyms to the word opinion are: belief, judgement, thought(s), school of thought, thinking, way of thinking, mind, point of view, view, viewpoint, outlook, angle, slant, side, attitude, stance, perspective, position, standpoint etc.
+
+It is interesting that the synonyms include belief, judgement and way of thinking. That is what our opinions are to us. Both meanings of the word are relevant to us as software developers. After all, we are a very opinionated bunch. Whether it’s Functional vs Object Oriented, Dynamic vs Static, Classicist vs Outside-in, AWS vs Azure, Mac vs Linux vs Windows, Star Trek vs Star Wars, we voice our opinion as the most valid, if not the absolute fact.
+
+“Opinionated” is not bad. Having none, or little, opinion is arguably ineffective. However, the validity of your opinion is extremely relevant. Is your opinion a mere gut-feel or a statement of fact? Opinion without qualification is irresponsible.
```
Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: Style of quotes: is “Opinionated” valid or do you need "Opinionated" ?

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+The word opinion has two distinct meanings:
+
+1. A view or judgement formed about something, not necessarily based on fact or knowledge.
+2. A statement of advice by an expert on a professional matter.
+
+Synonyms to the word opinion are: belief, judgement, thought(s), school of thought, thinking, way of thinking, mind, point of view, view, viewpoint, outlook, angle, slant, side, attitude, stance, perspective, position, standpoint etc.
+
+It is interesting that the synonyms include belief, judgement and way of thinking. That is what our opinions are to us. Both meanings of the word are relevant to us as software developers. After all, we are a very opinionated bunch. Whether it’s Functional vs Object Oriented, Dynamic vs Static, Classicist vs Outside-in, AWS vs Azure, Mac vs Linux vs Windows, Star Trek vs Star Wars, we voice our opinion as the most valid, if not the absolute fact.
```
Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: Classicist vs Outside-in what? styles of TDD

Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: very opinionated bunch: opinion - based on what?

Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: > Star Trek vs Star Wars,

Formatting:

> Star Trek vs Star Wars, ...

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+- craftsmanship
+
+---
+
+The word opinion has two distinct meanings:
+
+1. A view or judgement formed about something, not necessarily based on fact or knowledge.
+2. A statement of advice by an expert on a professional matter.
+
+Synonyms to the word opinion are: belief, judgement, thought(s), school of thought, thinking, way of thinking, mind, point of view, view, viewpoint, outlook, angle, slant, side, attitude, stance, perspective, position, standpoint etc.
```
Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md:20]: Please cite the source for the synonyms

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+We also have cross-cutting dimensions that make our opinions even weaker. For example, we may think that an opinion is based on an experience but the experience is second hand. Similarly, a ‘fact’ can also be passed second hand. You then must start thinking about the reliability of this second-hand source of fact or experience. 
+
+## What’s the point to all this?
+Qualifying our opinion according to its strength allows us to refine our thinking and the advice we give to others. If we have an opinion that is at the intuition level, then we can seek to strengthen it by seeking experiences and evidence. During this strengthening process, we may look at alternatives to our opinions and see if they are of a better strength.
+
+A qualified opinion, given as advice, provides the receiver more information and a better context in which to utilise that advice.
+
+Note: this post is also an opinion at the “intuition” level, deduced from my experiences. If you are interested in better formulating your thoughts and ideas then you may find the theory on [Critical Thinking](https://en.wikipedia.org/wiki/Critical_thinking) quite helpful.
```
Comment (alvarogarcia7) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: I'm missing the idea / reference of 'how much do I care about this', which mainly applies to decisions but could also apply to opinions

**Review by alvarogarcia7 — CHANGES_REQUESTED** [review]

File: _posts/2017-01-16-hierarchy-of-opinion.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+“Opinionated” is not bad. Having none, or little, opinion is arguably ineffective. However, the validity of your opinion is extremely relevant. Is your opinion a mere gut-feel or a statement of fact? Opinion without qualification is irresponsible.
+
+## Qualify your opinion
+So how do you qualify an opinion? People often start with “I think” or “In my opinion” when they’re expressing an opinion. Arguably, this kind of start for your opinion makes it explicit that you are expressing an opinion, but it is not enough. It does not convey the strength of your opinion. 
+
+In its strongest form, an opinion is based on a verified fact or empirical evidence. You can then express your opinion in the context of the fact or evidence.
+
+A somewhat weaker opinion could one formed from an experience. You can say, “from experience I have seen this to be true”. Just because you have experienced something does not mean that it is true in every case. You are assuming, that the current situation may be identical to the ones you have experienced before and that there are no variables.
```
Comment (Gryff) [inline _posts/2017-01-16-hierarchy-of-opinion.md]: could [be] one formed from an experience

File: _posts/2017-01-16-hierarchy-of-opinion.md (hunk 5 above)
Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: Not sure what you mean here. Can you explain more?

File: _posts/2017-01-16-hierarchy-of-opinion.md (hunk 4 above)
Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md:20]: source for definition and synonyms are the same - adding it twice is clumsy so no direct citation needed here.

File: _posts/2017-01-16-hierarchy-of-opinion.md (hunk 3 above)
Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: Made it more explicit.

Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: Added footnote.

Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: Can't see the difference.

File: _posts/2017-01-16-hierarchy-of-opinion.md (hunk 2 above)
Comment (mashooq) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: Changed the style of quotes.

File: _posts/2017-01-16-hierarchy-of-opinion.md (hunk 5 above)
Comment (alvarogarcia7) [reply _posts/2017-01-16-hierarchy-of-opinion.md]: I'm talking about this article, that we have discussed in one of our catchups. Please disregard the title: http://blog.capwatkins.com/the-sliding-scale-of-giving-a-fuck

It's about "how much do I care about the results of this decision". Does this also apply to "how much do I care about this opinion"?

**Comment by alvarogarcia7** [comment]
The thing about "the scale of..." is optional.

@mashooq Thanks for addressing my concerns

**Pull request merged**